cachito/web/api_v1.py

```diff
@@ -74,6 +77,31 @@ def get_requests():
         state_int = RequestStateMapping.__members__[state].value
         query = query.join(RequestState, Request.request_state_id == RequestState.id)
         query = query.filter(RequestState.state == state_int)
+    repo = flask.request.args.get("repo")
+    if repo:
+        query = query.filter(Request.repo == repo)
+    ref = flask.request.args.get("ref")
+    if ref:
+        if not is_request_ref_valid(ref):
+            raise ValidationError(f"{ref} is not a valid ref.")
+        query = query.filter(Request.ref == ref)
+    pkg_managers = flask.request.args.getlist("pkg_manager")
+    if pkg_managers:
+        pkg_manager_ids = []
+        for name in pkg_managers:
+            if not name:
+                # Ignore if pkg_manager= presents in the querystring
+                continue
+            pkg_manager: PackageManager = PackageManager.get_by_name(name)
+            if pkg_manager is None:
+                raise ValidationError(f"Cachito does not have package manager {name}.")
+            pkg_manager_ids.append(pkg_manager.id)
+        query = (
+            query.join(PackageManager, Request.pkg_managers)
+            .filter(PackageManager.id.in_(pkg_manager_ids))
```

Comment:
Could we do an outer join and allow `PackageManager.id` to be NULL in the filter? That way, we could search for requests that do not use any package managers with `/api/v1/requests?pkg_manager=`

Reply:
(or just do 2 different queries, this one if `pkg_manager_ids` is not empty, outer join and select NULL if `pkg_manager_ids` is empty)